File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/tasks/offlinemap/OfflineMapSyncJobNativeObject.java
package com.valentingrigorean.arcgis_maps_flutter.tasks.offlinemap;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.esri.arcgisruntime.tasks.geodatabase.GeodatabaseDeltaInfo;
import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapSyncJob;
import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapSyncResult;
import com.esri.arcgisruntime.util.ListenableList;
import com.valentingrigorean.arcgis_maps_flutter.concurrent.JobNativeHandler;
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.BaseNativeObject;
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.NativeHandler;
import com.valentingrigorean.arcgis_maps_flutter.io.RemoteResourceNativeHandler;

import java.util.ArrayList;

import io.flutter.plugin.common.MethodChannel;

public class OfflineMapSyncJobNativeObject extends BaseNativeObject<OfflineMapSyncJob> {
    public OfflineMapSyncJobNativeObject(String objectId, OfflineMapSyncJob job) {
        super(objectId, job, new NativeHandler[]{
                new JobNativeHandler(job, JobNativeHandler.JobType.OFFLINE_MAP_SYNC),
                new RemoteResourceNativeHandler(job),
        });
    }

    @Override
    public void onMethodCall(@NonNull String method, @Nullable Object args, @NonNull MethodChannel.Result result) {
        switch (method) {
            case "offlineMapSyncJob#getGeodatabaseDeltaInfos": {
                final ListenableList<GeodatabaseDeltaInfo> geodatabaseDeltaInfos = getNativeObject().getGeodatabaseDeltaInfos();
                final ArrayList<Object> list = new ArrayList<>(geodatabaseDeltaInfos.size());
                for (GeodatabaseDeltaInfo geodatabaseDeltaInfo : geodatabaseDeltaInfos) {
                    list.add(ConvertOfflineMap.geodatabaseDeltaInfoToJson(geodatabaseDeltaInfo));
                }
                result.success(list);
            }
            break;
            case "offlineMapSyncJob#getResult": {
                final OfflineMapSyncResult syncResult = getNativeObject().getResult();
                if (syncResult != null) {
                    result.success(ConvertOfflineMap.offlineMapSyncResultToJson(syncResult));
                } else {
                    result.success(null);
                }
            }
            break;
            default:
                super.onMethodCall(method, args, result);
                break;
        }
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/tasks/offlinemap/OfflineMapSyncJobNativeObject.kt
package com.valentingrigorean.arcgis_maps_flutter.tasks.offlinemap

import com.arcgismaps.tasks.offlinemaptask.OfflineMapSyncJob
import com.valentingrigorean.arcgis_maps_flutter.concurrent.JobNativeHandler
import com.valentingrigorean.arcgis_maps_flutter.convert.tasks.offlinemap.ConvertOfflineMap
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.BaseNativeObject
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.NativeHandler
import io.flutter.plugin.common.MethodChannel

class OfflineMapSyncJobNativeObject(objectId: String, job: OfflineMapSyncJob) :
    BaseNativeObject<OfflineMapSyncJob>(
        objectId, job, arrayOf<NativeHandler>(
            JobNativeHandler(job),
        )
    ) {
    override fun onMethodCall(method: String, args: Any?, result: MethodChannel.Result) {
        when (method) {
            "offlineMapSyncJob#getGeodatabaseDeltaInfos" -> {
                val geodatabaseDeltaInfos = nativeObject.getGeodatabaseDeltaInfos()
                val list = ArrayList<Any?>(geodatabaseDeltaInfos.size)
                for (geodatabaseDeltaInfo in geodatabaseDeltaInfos) {
                    list.add(ConvertGeodatabase.geodatabaseDeltaInfoToJson(geodatabaseDeltaInfo))
                }
                result.success(list)
            }

            "offlineMapSyncJob#getResult" -> {
                val syncResult = nativeObject.getResult()
                if (syncResult != null) {
                    result.success(ConvertOfflineMap.offlineMapSyncResultToJson(syncResult))
                } else {
                    result.success(null)
                }
            }

            else -> super.onMethodCall(method, args, result)
        }
    }
}